src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ConsoleDebugTargetsProvider.cs

```diff
@@ -0,0 +1,322 @@
+    {
+
+        [ImportingConstructor]
+        public ConsoleDebugTargetsProvider(ConfiguredProject configuredProject, 
+                                           IDebugTokenReplacer tokenReplacer, 
+                                           IFileSystem fileSystem, 
+                                           IEnvironmentHelper environment,
+                                           ProjectProperties properties)
+        {
+            ConfiguredProject = configuredProject;
+            TokenReplacer = tokenReplacer;
+            TheFileSystem = fileSystem;
+            TheEnvironment = environment;
+            Properties = properties;
+        }
+
+        private ConfiguredProject ConfiguredProject { get; }
+        private IDebugTokenReplacer TokenReplacer { get; }
+        private IFileSystem TheFileSystem { get; }
+        private IEnvironmentHelper TheEnvironment { get; }
+        private ProjectProperties Properties { get; }
+
+        /// <summary>
+        /// This provider handles running the Project and empty empty commandName (this generally just runs the executable)
+        /// </summary>
+        public bool SupportsProfile(ILaunchProfile profile)
+        {
+            return string.IsNullOrWhiteSpace(profile.CommandName) || profile.CommandName.Equals(LaunchSettingsProvider.RunProjectCommandName, StringComparison.OrdinalIgnoreCase);
+        }
+
+        /// <summary>
+        /// Called just prior to launch to do additionl work (put up ui, do special configuration etc).
+        /// </summary>
+        public Task OnBeforeLaunchAsync(DebugLaunchOptions launchOptions, ILaunchProfile profile)
+        {
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Called just after the launch to do additionl work (put up ui, do special configuration etc).
+        /// </summary>
+        public Task OnAfterLaunchAsync(DebugLaunchOptions launchOptions, ILaunchProfile profile)
+        {
+            return Task.CompletedTask;
+        }
+
+        private async Task<bool> GetIsClassLibraryAsync()
+        {
+            // Used by default Windows debugger to figure out whether to add an extra
+            // pause to end of window when CTRL+F5'ing a console application
+            var configuration = await Properties.GetConfigurationGeneralPropertiesAsync()
+                                                 .ConfigureAwait(false);
+
+
+            IEnumValue outputType = (IEnumValue)await configuration.OutputType.GetValueAsync()
+                                                                      .ConfigureAwait(false);
+
+            return StringComparers.PropertyValues.Equals(outputType.Name, ConfigurationGeneral.OutputTypeValues.Library);
+        }
+
+        /// <summary>
+        /// This is called on F5\Ctrl-F5 to return the list of debug targets.What we return depends on the type
+        /// of project.  The dth port contains the port of the design time host. The cmdArgs will only be 
```

Comment:
"The dth port contains the port of the design time host."

I read the method, and I have no idea what this comment signifies. What is a port? What's the dth of them?